File: Question1/pow.cpp
#include <iostream>
#include <math.h>
using namespace std;



int powRecursive(int b, int p)
{
	if (p == 0)
	{
		return 1;
	}
	else if (p == 1)
	{
		return b;
	}
	else
	{
		return b * powRecursive(b, p - 1);
	}

}

int powIterative(int b, int p)

{
	int result= 1;
	int i = 0;

	while (i < p)
	{
		result *= b ;
		i++;
	}

	return result;
}





int main()
{

	int b, p;

	cin >> b;
	cin >> p;

	cout << "Using Pow " << pow(b, p) << endl;

	cout <<  "Recursive " << powRecursive(b, p) << endl;

	cout << "Iterative " << powIterative(b, p) << endl;
}

File: Question1/pow.py
import os

def main():
	
	base = int(input())
	power = int(input())

	output = powSimple(base, power)

	print(output)


	output2 = powIterative(base, power);
	print("Iterative " + str(output2))c


def powSimple(base , power):
	return base ** power

def powIterative(base, power):
	i = 0
	result = 1

	for i in range (power):
		result *= base;

	return result

main()